cmd/plugins/balloons/policy/balloons-policy.go

```diff
@@ -1202,6 +1206,9 @@ func (p *balloons) fillCloseToDevices(blnDefs []*BalloonDef) {
 		if blnDef.PreferIsolCpus {
 			blnDef.PreferCloseToDevices = append(blnDef.PreferCloseToDevices, virtDevIsolatedCpus)
 		}
+		if cpuset.MustParse(blnDef.PreferCpus).Size() != 0 {
```

Comment:
There is no error handling for invalid preferCpus, and the value is parsed many times. The opposite would be better.

Would you change this so that you parse each preferCpus only once and store the value to an internal attribute of balloon type? If you'd do this as part of configuration validation, then you could give a nice error message on parse errors.

Reply:
`MustParse` doesn't return an error so I guess moving the caller to validateConfig won't change anything in terms of error handling. But, I think we can use `cpuset.Parse()` similarly to [this](https://github.com/containers/nri-plugins/blob/46d22a67f989bc0fefe679e532461429a13936d1/pkg/utils/cpuset/cpuset.go#L35)?

Reply:
and I'm okay to move the calling the MustParse/Parse to `validateConfig()` .

Reply:
But probably we should all agree first if this PR is something that can be beneficial and merged. Because otherwise I will waste my time trying to fix something that we don't want to take in.

Reply:
> `MustParse` doesn't return an error so I guess moving the caller to validateConfig won't change anything in terms of error handling. But, I think we can use `cpuset.Parse()` similarly to [this](https://github.com/containers/nri-plugins/blob/46d22a67f989bc0fefe679e532461429a13936d1/pkg/utils/cpuset/cpuset.go#L35)?

Yes, exactly. We definitely do not want to crash on a typo in a config.

> But probably we should all agree first if this PR is something that can be beneficial and merged. Because otherwise I will waste my & your time trying to fix something that we don't want to take in.

I agree. I think this PR is a very good start for this feature, thank you for writing it! Let's continue if @klihub and @kad show green(ish) light for this. I think you can put this PR into the draft state.

Would you proceed with preferCoreType (or preferPCores/preferECores... feel free to decide what would be the most suitable parameter name) instead? You should be able to test it in our e2e vm by adding environment variables to balloons policy container: look for OVERRIDE_SYS_CORE_CPUS and OVERRIDE_SYS_ATOM_CPUS from the code. Those variables enable faking that there are P and E cores in the system.